Code/Framework/AzGameFramework/AzGameFramework/Application/GameApplication.cpp

```diff
@@ -12,8 +12,13 @@
 #include <AzCore/Utils/Utils.h>
 
 #include <AzFramework/Archive/Archive.h>
+#include <AzFramework/Components/NativeUISystemComponent.h>
 #include <AzGameFramework/AzGameFrameworkModule.h>
 
+#if !LY_HEADLESS
+#include <AzFramework/AzFrameworkNativeUIModule.h>
+#endif
```

Comment:
I recommend using O3DE for any new CMake Cache variable names and defines.
I also recommend think the define of HEADLESS is too TERSE and should have _SERVER added to it.

```suggestion
#if !O3DE_HEADLESS_SERVER
#include <AzFramework/AzFrameworkNativeUIModule.h>
#endif
```